core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -120,6 +120,7 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Ex
                     final UnprocessedRequestException wrappedCause = UnprocessedRequestException.of(cause);
                     handleEarlyRequestException(ctx, req, wrappedCause);
                     res.close(wrappedCause);
+                    ctx.cancel(cause);
```

Comment:
(DNS timeout case)

Reply:
```
requestCtx.close();
requestCtx.cancel(timeoutException);
// to set isTimedout, we need ctx.cancel. should we ctx.cancel after/before ctx.close() ? 🤔
```
Hmm overall code, I'm not sure proper order of ctx.cancel(cause).

Also, I think it's hard to test this by unit test. is there any good idea, please share to me~!

Reply:
- I prefer to call `ctx.cancel()` first because `whenResponseCancelled()` should be invoked before a response is cancelled.
- What do you think of using `wrappedCause` as it for consistency? We can override the `RequestContext.isTimedOut()` in `ClientRequestContext` to unwrap `UnprocessedRequestException`.
- Would you mind adding tests for the changes?

Reply:
We can use `TestDnsServer` to trigger DNS timeout.
https://github.com/line/armeria/blob/dbe724ab3fb9a4cae8ad9d981c5334b1ef4a9fcd/core/src/test/java/com/linecorp/armeria/client/endpoint/dns/TestDnsServer.java#L52-L53

Reply:
✅ updated!